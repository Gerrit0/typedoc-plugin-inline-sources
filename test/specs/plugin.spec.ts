import * as path from 'path';
import { Application, DeclarationReflection, ProjectReflection } from 'typedoc';

describe(`Options 1:`, () => {
  let app: Application;
  let project: ProjectReflection;

  beforeAll(async () => {
    app = await Application.bootstrapWithPlugins({
      options: path.join(__dirname, '../typedoc.json'),
    });
    project = (await app.convert()) as ProjectReflection;
    console.log(project.children?.length);
  });

  test(`should inject variable source`, async () => {
    const reflection = project.getChildByName('someVariable');
    expect(reflection?.comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject variable source for variable list (1)`, async () => {
    const reflection1 = project.getChildByName('multipleVariable1');
    expect(reflection1?.comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject variable source for variable list (2)`, async () => {
    const reflection2 = project.getChildByName('multipleVariable2');
    expect(reflection2?.comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject type alias source`, async () => {
    const reflection = project.getChildByName('SomeType');
    expect(reflection?.comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject function source`, async () => {
    const reflection = project.getChildByName(
      'someFunction',
    ) as DeclarationReflection;
    expect(reflection.signatures![0].comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject class source`, async () => {
    const reflection = project.getChildByName('SomeClass');
    expect(reflection?.comment?.blockTags).toMatchSnapshot();
  });

  test(`should inject interface source`, async () => {
    const reflection = project?.getChildByName('SomeInterface');
    expect(reflection?.comment?.blockTags).toMatchSnapshot();
  });
});
