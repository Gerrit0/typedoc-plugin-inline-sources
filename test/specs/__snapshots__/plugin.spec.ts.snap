// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options 1: should inject class source 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export class SomeClass {
  /**
   * Some prop comments
   */
  someProp: string;

  // Class method source
  someMethod() {
    return 'hello';
  }
}
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject function source 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export function someFunction(x: number, y: number) {
  return x + y;
}
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject interface source 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export interface SomeInterface {
  /**
   * Comments for prop1
   */
  prop1: string;

  // Comments for prop2
  prop2: number;
}
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject type alias source 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export type SomeType = number;
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject variable source 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "text",
        "text": "Some source comments

Some more source comments",
      },
      {
        "kind": "code",
        "text": "

\`\`\`ts
export const someVariable = /*ok*/ 'hello'; // someVariable comment;
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject variable source for variable list (1) 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export const multipleVariable1 = 'hello-1',
  multipleVariable2 = 'hello-2';
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;

exports[`Options 1: should inject variable source for variable list (2) 1`] = `
[
  CommentTag {
    "content": [
      {
        "kind": "code",
        "text": "

\`\`\`ts
export const multipleVariable1 = 'hello-1',
  multipleVariable2 = 'hello-2';
\`\`\`

",
      },
    ],
    "skipRendering": false,
    "tag": "@source",
  },
]
`;
